### PR TITLE
Improve missing cassette errors

### PR DESCRIPTION
**Pull request by therve**
When the freeze file or the cassette file is missing, we get pretty
weird errors. Let's try to make them more explicit.

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -211,24 +209,16 @@ def context(vcr, unique, unique_lower, freezer):
...
+        os.getenv("RECORD", "false").lower()
+    ]
 
 
 def _disable_recording():
     """Disable VCR.py integration."""
     return os.getenv("RECORD", "false").lower() == "none"
 
 
-@pytest.fixture(scope="session")
```
Comment (NouemanKHAL) [inline tests/conftest.py:226]: Why are we removing this ?

Comment (therve) [reply tests/conftest.py:226]: It's not used anywhere, and I'm not sure what it does.

Comment (NouemanKHAL) [reply tests/conftest.py:226]: It's a redefinition of a builtin fixture: https://github.com/kiwicom/pytest-recording/blob/master/src/pytest_recording/plugin.py#L74

Comment (therve) [reply tests/conftest.py:226]: Do you think we need it?

Comment (therve) [reply tests/conftest.py:226]: Good catch!

**Comment by NouemanKHAL** [comment]
When a cassette is missing it still prints `vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ..`

A potential fix that seems to work for me locally is to move this block:

```
 except CannotOverwriteExistingCassetteException as e:
        if os.getenv("RECORD", "false").lower() == "false":
            msg = (
                "Cassette '{}' not found: create one setting `RECORD=true` or "
                "ignore it using `RECORD=none`".format(e.cassette._path)
            )
            raise RuntimeError(msg) from e
        raise e
```

to line 528.

**Comment by therve** [comment]
> When a cassette is missing it still prints `vcr.errors.CannotOverwriteExistingCassetteException: Can't overwrite existing cassette ..`
> 
> A potential fix that seems to work for me locally is to move this block:
> 
> ```
>  except CannotOverwriteExistingCassetteException as e:
>         if os.getenv("RECORD", "false").lower() == "false":
>             msg = (
>                 "Cassette '{}' not found: create one setting `RECORD=true` or "
>                 "ignore it using `RECORD=none`".format(e.cassette._path)
>             )
>             raise RuntimeError(msg) from e
>         raise e
> ```
> 
> to line 528.

I'm not exactly sure what you mean by that? It prints it on purpose because of the "from e" in the raise.

**Review by NouemanKHAL — APPROVED** [review]

**Pull request merged**